src/osp/Universe.h

```diff
@@ -173,6 +183,14 @@ TYPESAT_T& Universe::type_register(ARGS_T&& ... args)
     return toReturn;
 }
 
+template<typename SATTYPE_T>
+SATTYPE_T& Universe::sat_type_find()
+{
+    MapSatType::iterator it = sat_type_find(SATTYPE_T::smc_name);
+    assert(it != m_satTypes.end());
+    return static_cast<SATTYPE_T&>(*it->second);
```

Comment:
assert on a dynamic cast as well. I provided an example in another comment.

Reply:
I can't find any comment with dynamic_cast example code. Again, RTTI is not used anywhere. The only time this will fail is if there is two different satellite types with the same name constant, which can only happen if the program was compiled that way. There is also the possibility of using entt's unique class naming functions instead of string constants.

Reply:
> The only time this will fail is if there is two different satellite types with the same name constant, which can only happen if the program was compiled that way.

How can you statically assert that to ensure it's not possible?

> Again, RTTI is not used anywhere

Is RTTI disabled as a cli option to the compiler?